admin/test/test_installer.py

```diff
@@ -0,0 +1,103 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+"""
+Tests for ``admin.installer``.
+"""
+from textwrap import dedent
+
+from pyrsistent import PClass, field, pmap_field
+
+from flocker.testtools import (
+    TestCase
+)
+
+from ..installer._images import (
+    _packer_amis, _PackerOutputParser,
+)
+
+
+class ParserData(PClass):
+    input = field(type=unicode, mandatory=True)
+    output = pmap_field(key_type=unicode, value_type=unicode, optional=False)
+
+
+PACKER_OUTPUT_US_WEST_1 = ParserData(
+    input=u"""\
+1450420216,,ui,say,Build 'amazon-ebs' finished.
+1450420216,,ui,say,\n==> Builds finished. \
+The artifacts of successful builds are:
+1450420216,amazon-ebs,artifact-count,1
+1450420216,amazon-ebs,artifact,0,builder-id,mitchellh.amazonebs
+1450420216,amazon-ebs,artifact,0,id,us-west-1:ami-e098f380
+1450420216,amazon-ebs,artifact,0,string,AMIs were created:\n\n\
+us-west-1: ami-e098f380
+1450420216,amazon-ebs,artifact,0,files-count,0
+1450420216,amazon-ebs,artifact,0,end
+1450420216,,ui,say,--> amazon-ebs: AMIs were created:\n\n\
+us-west-1: ami-e098f380
+""",
+    output={u"us-west-1": u"ami-e098f380"}
+)
+
+PACKER_OUTPUT_US_ALL = ParserData(
+    input=u"""\
```

Comment:
:grey_question: I don't follow the use of backslashes here. Are they needed and the end of some lines, but not others?